libevmasm/Assembly.cpp

```diff
@@ -1281,6 +1282,21 @@ LinkerObject const& Assembly::assembleLegacy() const
 	uint8_t tagPush = static_cast<uint8_t>(pushInstruction(bytesPerTag));
 	uint8_t dataRefPush = static_cast<uint8_t>(pushInstruction(bytesPerDataRef));
 
+	LinkerObject::CodeSectionLocation codeSectionLocation;
+	codeSectionLocation.start = 0;
+	size_t assemblyItemIndex = 0;
+	std::function assembleInstruction = [&](std::function<void()> const& _addInstruction){
```

Comment:
```suggestion
	auto assembleInstruction = [&](auto&& _addInstruction){
```
Not sure if this was discussed - but I'd really use ``auto`` here - ``std::function`` is not the right type and does have overhead that is hard to optimize away - and we're doing this a lot below.
Functionally, there's no difference, though, so I'd not stop merging this due to this.

Reply:
I mean, I'd not even make it take a function. All it needs is a `bytes` array containing a newly added piece of bytecode. A function is kinda overkill for this and this style of doing things just reminds me of JS callback hell :)